src/snowflake/snowpark/functions.py

```diff
@@ -1250,6 +1284,27 @@ def floor(e: ColumnOrName) -> Column:
     return builtin("floor")(c)
 
 
+def format_number(col: ColumnOrName, d: int):
+    """Format numbers to a specific number of decimal places.
+    Example::
+            >>> from snowflake.snowpark.functions import format_number
+            >>> data = [(1, 3.14159),
+            ...(2, 2.71828),
+            ...(3, 1.41421)]
+            >>> df = session.createDataFrame(data, ["id", "value"])
+            >>> df.select("id",format_number("value",2).alias("value")).show()
+            ----------------------------------
+            |"ID"  |"VALUE"                  |
+            ----------------------------------
+            |1     |                   3.14  |
+            |2     |                   2.72  |
+            |3     |                   1.41  |
+            ----------------------------------
+    """
+    col = _to_col_if_str(col, "format_number")
+    return to_varchar(col, "999,999,999,999,999." + "0" * d)
```

Comment:
this looks like a workaround. Have you tried using existing methods such as `.cast(DecimalType(x, y))`?